src/brod_producer.erl

```diff
@@ -434,15 +442,15 @@ maybe_demonitor(Mref) ->
 %% @private
 schedule_retry(#state{buffer = Buffer} = State, Reason) ->
   {ok, NewBuffer} = brod_producer_buffer:nack_all(Buffer, Reason),
```

Comment:
I feel it will be more clear if we call `nack_all` explicitly where it belongs and have only one version of `schedule_retry` so that the function does exactly what the name implies.